src/ObjectMultiplex.ts

```diff
@@ -0,0 +1,93 @@
+import { Duplex } from 'stream';
+import eos from 'end-of-stream';
+import once from 'once';
+import { Substream } from './Substream';
+
+const IGNORE_SUBSTREAM: Record<string, unknown> = {};
+
+export class ObjectMultiplex extends Duplex {
+
+  private _substreams: Record<string, Substream | Record<string, any>>;
+
+  constructor(_opts: Record<string, unknown> = {}) {
+    const opts = Object.assign({}, _opts, {
+      objectMode: true,
+    });
+    super(opts);
+    this._substreams = {};
+  }
+
+  createStream(name: string): Substream {
+    // validate name
+    if (!name) {
+      throw new Error('ObjectMultiplex - name must not be empty');
+    }
+
+    if (this._substreams[name]) {
+      throw new Error(`ObjectMultiplex - Substream for name "${name}" already exists`);
+    }
+
+    // create substream
+    const substream = new Substream({ parent: this, name });
+    this._substreams[name] = substream;
+
+    // listen for parent stream to end
+    anyStreamEnd(this, (_error?: Error | null) => {
+      return substream.destroy();
```

Comment:
In the original javascript code, **_destroy()_** method is called by passing _error to it, when I introduce the types, there is a mismatch in **error** type accepted by **_eos(stream, { readable: false }, cb)_** in the function definition and **_substream.destroy()_**

**eos** accepts _Error|null|undefined_ whereas **destroy** accepts only _Error|undefined_, because of this confusion I did not pass the _error to subStream.destroy() call, there might be a chance that this is causing the current error but I am confused on what type to use otherwise.

Reply:
Oh, I just saw this. I think we should continue to pass the error, and we can do it like this:
```
  return substream.destroy(_error || undefined);
```